### ISSUE TITLE
Constraint dynamical addition fails

### ISSUE THREAD

**Issue by jgFages** [chocoteam/choco3#277]
See testCostas of DynamicPostTest

**Comment by jgFages** [comment]
This happens with some global constraints having an internal data structure that is initialized at constraint creation (e.g. alldiff AC build a bipartite graph with a value mapping).

=> If the constraint is created at the middle of the search, then these data structures are initialized with the current domains. Therefore, it does not handle values that had been filtered previously. This means a problem may occur upon backtrack (when these values come back into the domains). And this is somehow normal.

In conclusion, we may need to review global constraints to identify which ones cannot be dynamically added

**Comment by cprudhom** [comment]
Fixed in 3.3.2
